rs/sns/swap/src/swap.rs

```diff
@@ -894,9 +894,24 @@ impl Swap {
         // for all NF investors.
         let mut global_cf_memo: u64 = NEURON_BASKET_MEMO_RANGE_START;
         for cf_participant in self.cf_participants.iter_mut() {
+            let controller = cf_participant.try_get_controller();
+
             for cf_neuron in cf_participant.cf_neurons.iter_mut() {
                 // Create a closure to ensure `global_cf_memo` is incremented in all cases
                 let mut process_cf_neuron = || {
+                    let controller = match controller.clone() {
+                        Ok(nns_neuron_controller_principal) => nns_neuron_controller_principal,
+                        Err(e) => {
+                            log!(
+                                ERROR,
+                                "Error getting the controller for {cf_neuron:?} principal: {e}"
+                            );
+                            sweep_result.invalid +=
+                                neuron_basket_construction_parameters.count as u32;
+                            return;
+                        }
+                    };
```

Comment:
Note to reviewer: this check must be done inside this loop because naively moving it outside of the loop would change how many times we increment sweep_result.invalid. (Unless we did something more complicated I guess). And if we're doing it inside the loop, we should do it in the closure I think, otherwise it makes the control flow complicated